src/components/services/AccountCallbackManager.tsx

```diff
@@ -90,6 +92,16 @@ export function AccountCallbackManager(props: Props) {
         dispatch(refreshTransactionsRequest(wallet, transactions))
         dispatch(newTransactionsRequest(navigation, wallet, transactions))
 
+        // Check for incoming FIO requests:
+        const receivedTxs = transactions.filter(tx => !tx.isSend)
+        if (receivedTxs.length > 0) dispatch(checkFioObtData(wallet, receivedTxs))
```

Comment:
Just realized there is a race condition here that predates your changes. `checkFioObtData` might change the EdgeTransaction as we are showing the dropdown and then show tx details. We should be passing txids instead of tx objects to the txdetails scene and add `getTransactionById` call into the plugins. TxDetails can get the most up to date version of the EdgeTransaction. 

Or for now, have checkFioObtData actually mutate the EdgeTransaction in the transactions array so it shows properly in Tx Details.

Noting this for a future PR.